at_response.c

```diff
@@ -497,9 +496,28 @@ static int at_response_error (struct pvt* pvt, at_res_t res)
 				pvt->outgoing_sms = 0;
 				pvt_try_restate(pvt);
 
-				manager_event_sent_notify(PVT_ID(pvt), "SMS", task, "NotSent");
+				{
+					char payload[4096], dst[256];
```

Comment:
You're passing dst and payload to smsdb where you're writing to them, without knowing anything about the passed sizes. Those hardcoded values won't do.